src/cli.rs

```diff
@@ -0,0 +1,35 @@
+use clap::{Parser, Subcommand};
+use lndk::lndk_offers::decode;
+
+/// A cli for interacting with lndk
+#[derive(Debug, Parser)]
+#[command(name = "lndk-cli")]
+#[command(about = "A cli for interacting with lndk", long_about = None)]
+struct Cli {
+    #[command(subcommand)]
+    command: Commands,
+}
+
+#[derive(Debug, Subcommand)]
+enum Commands {
+    /// Decodes a bech32-encoded offer string into a BOLT 12 offer
+    Decode {
+        /// The offer string to decode
+        offer_string: String,
+    },
+}
+
+fn main() {
+    let args = Cli::parse();
+    match args.command {
+        Commands::Decode { offer_string } => {
+            println!("Decoding offer: {offer_string}");
+            match decode(offer_string) {
+                Ok(offer) => println!("Decoded offer: {:?}", offer),
+                Err(_) => {
+                    println!("ERROR provided offer argument is invalid")
```

Comment:
Is this error meaningful enough to print?

Reply:
Oh my bad for missing this. I think so because it tells the user that the provided offer string is not what we need. Maybe we could add a line like "Make sure you're inputting a bech32 offer string that starts in lno..." Ultimately this is the error enum that's being returned: https://docs.rs/lightning/latest/lightning/offers/parse/enum.Bolt12ParseError.html We could provide more specifics on the enum variant that was thrown but feeling like those specifics might not be helpful for most users. Thoughts?

Reply:
What about a combo of the two?
"Please provide offer lno.., failed to decode with error: X"

While the detail isn't useful for the end user, it's probably useful for the developer who they go complain to about the invalid offer?